praw/models/reddit/redditor.py

```diff
@@ -123,14 +123,22 @@ def __init__(self, reddit, name=None, fullname=None, _data=None):
             )
         if _data:
             assert (
-                isinstance(_data, dict) and "name" in _data
+                    isinstance(_data, dict) and "name" in _data
             ), "Please file a bug with PRAW"
         super(Redditor, self).__init__(reddit, _data=_data)
         self._listing_use_sort = True
         if name:
             self.name = name
+            if not isinstance(name, str):
+                raise TypeError(
+                    "The name must be type `str`."
+                )
         elif fullname:
             self._fullname = fullname
+            if not isinstance(fullname, str):
+                raise TypeError(
+                    "The id must be type `str`."
```

Comment:
Message should say `fullname` rather than `id`

Reply:
fixed it now, thanks